airflow-core/src/airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -308,6 +310,19 @@ def ti_update_state(
         updated_state = ti_patch_payload.state
         query = TI.duration_expression_update(ti_patch_payload.end_date, query, session.bind)
         query = query.values(state=updated_state)
+
+        if updated_state == TerminalTIState.FAILED:
+            ti = session.get(TI, ti_id_str)
+            row = session.query(SerializedDagModel).filter_by(dag_id=dag_id).first()
```

Comment:
This isn't quite right -- you'll need to get it out of the dag bag, and see #48956 as we'll need to get the right version of it out.

Reply:
Yeah let me try and do it using dagbag